engines/advancedDetector.cpp

```diff
@@ -187,17 +187,26 @@ DetectedGame AdvancedMetaEngineDetection::toDetectedGame(const ADDetectedGame &a
 	}
 
 	DetectedGame game(getName(), desc->gameId, title, desc->language, desc->platform, extra, ((desc->flags & (ADGF_UNSUPPORTED | ADGF_WARNING)) != 0));
-	game.hasUnknownFiles = adGame.hasUnknownFiles;
+	game.hasUnknownFiles = adGame.getHasUnknownFiles();
 	game.matchedFiles = adGame.matchedFiles;
 
+	Common::HashMap<Common::String, bool, Common::IgnoreCase_Hash, Common::IgnoreCase_EqualTo> usesResourceFork;
+
+	for (const ADGameFileDescription *fileDesc = desc->filesDescriptions; fileDesc->fileName; fileDesc++) {
+		if (fileDesc->md5[0] == 'r' && fileDesc->md5[1] == ':') {
+			usesResourceFork[fileDesc->fileName] = true;
+			break;
+		}
+	}
+
 	// Now specify the computation method for each file entry.
 	// TODO: This could be potentially overridden by use of upper part of adGame.fileType
 	// so, individual files could have their own computation method
 	for (FilePropertiesMap::iterator file = game.matchedFiles.begin(); file != game.matchedFiles.end(); ++file) {
-		if (desc->flags & ADGF_MACRESFORK)
-			file->_value.md5prop = (MD5Properties)(file->_value.md5prop | kMD5MacResFork);
 		if (desc->flags & ADGF_TAILMD5)
 			file->_value.md5prop = (MD5Properties)(file->_value.md5prop | kMD5Tail);
+		if (usesResourceFork[file->_key])
+			file->_value.md5prop = (MD5Properties)(file->_value.md5prop | kMD5MatchedByMacResFork);
```

Comment:
I think the verb 'Matched' is redundant here because everything in these flags is about matching.